src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1292,22 +1296,25 @@ int XrdHttpReq::ProcessHTTPReq() {
             if ((offs >= filesize) || (offs+l > filesize)) {
               TRACE(ALL, " Requested range " << l << "@" << offs <<
               " is past the end of file (" << filesize << ")");
-              //prot->SendSimpleResp(522, NULL, NULL, (char *) "Invalid range request", 0);
-              return -1;
+              httpStatusCode = 552;
```

Comment:
Maybe you can put the error message in a string allowing you to not repeat the same string in the two calls?

Reply:
Done in the latest commit.  Note the `httpStatusCode` in the commit (552) is different from the original one (522).

Both are wrong.  Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses, it should be `416 Range Not Satisfiable`.